### DISCUSSION TITLE
"Unklare" Entitäts-ID's und unvollständige Entitäts-Namen

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#1316] [Q&A] [answered]
Mir sind gestern meine HB-UNISensoren auf der Zentrale "abhanden" gekommen. Ich hatte das nicht gleich bemerkt und nebenbei HA geupdated.
Dadurch waren sie dann auch in HA (nach dessen Neustart) nicht mehr vorhanden.
Durch Einspielen eines Backups auf der Zentrale und einem HA Neustart war dann fast alles wieder wie vorher.

Fast, weil sich die Entitätsnamen bei den meisten Datenpunkten leicht geändert hatten.
von original:
`sensor.klima_sensor_device_bx_unisens003_klima_sensor_bx_unisens003_1_klimadaten_temperature`
zu deutsch:
`sensor.klima_sensor_device_bx_unisens003_klima_sensor_bx_unisens003_1_klimadaten_temperatur`

Das ist soweit ok, weil die Übersetzung des Datenpunkt-Namens wohl dem aktuellen Standard entspricht.

Nun gibt es aber noch 2 Datenpunkte bei denen das irgendwie nicht sauber klappt.
"Taupunkt" von original --> zu:
```
sensor.klima_sensor_device_bx_unisens003_klima_sensor_bx_unisens003_1_klimadaten_taupunkt
sensor.klima_sensor_device_bx_unisens003
```
"Abs_Luftfeuchte" von original --> zu:
```
sensor.klima_sensor_device_bx_unisens003_klima_sensor_bx_unisens003_1_klimadaten_abs_luftfeuchte
sensor.klima_sensor_device_bx_unisens003_2
```

![HA_Unisensor_Prob_Entity-ID](https://github.com/danielperna84/hahomematic/assets/54898249/aa3e3090-66f2-43a9-b0b2-177ff3cbc127)
![HA_Unisensor_Prob_Entity-ID_Zustand](https://github.com/danielperna84/hahomematic/assets/54898249/718c9362-72a4-430c-b1ea-61656260bd65)

Die Ansicht der Datenpunkte der Geräte (auf der Zentrale) ist hier zu sehen: https://github.com/danielperna84/hahomematic/discussions/276#discussioncomment-2128844

Die Frage wäre jetzt wo das hakt und ob du da ggf. was machen kannst.

**Comment by SukramJ** [comment]
Häng doch bitte mal einen Screenshot vom geöffneten Gerät aus der CCU an.

**Reply by Baxxy13** [reply]
Einstellungen > Geräte:
![HA_Unisensor_Prob_Entity-ID_Ansicht_WebUI](https://github.com/danielperna84/hahomematic/assets/54898249/0f8d1ddc-c3f7-4ae7-8a25-577ae4984f3d)

Status und Bedienung > Geräte:
![HA_Unisensor_Prob_Entity-ID_Ansicht_WebUI_Bedienung](https://github.com/danielperna84/hahomematic/assets/54898249/58dcdb28-e47a-4b96-ada3-7f34d6bdf51d)

Das meine "WebUI-Benamung" ungünstig ist weil dadurch die Entity-ID's und Entity-Namen so lang und "gedoppelt" werden ist mir schon vor einiger Zeit aufgefallen. Damit kann ich aber leben und ich möchte ungern meine Namensschema komplett ändern.

Hat es vielleicht was mit der mit der, bei den beiden Datenpunkten, nicht nötigen/ nicht möglichen Übersetzung zu tun?

**Reply by SukramJ** [reply]
Also die Namen wären schon mal kürzer wenn deine Kanäle mit  dem Gerätenamen (`Klima Sensor Device BX: UNISENS003`)beginnen würden.

**Reply by SukramJ** [reply]
> Hat es vielleicht was mit der mit der, bei den beiden Datenpunkten, nicht nötigen/ nicht möglichen Übersetzung zu tun?

Das denke ich nicht

**Reply by Baxxy13** [reply]
Wie geschrieben, mit der Länge komme ich klar. 
Das bei beginnender Namensgleichheit von Gerät und Kanal das weggelassen wird weiß ich.
Mir geht's aber nur um die beiden Problemdatenpunkte.

Ich kann aber mal probieren was bei raus kommt. 

Gesagt getan, nicht wirklich besser:
![HA_Unisensor_Prob_Entity-ID_Zustand_anderer_Name](https://github.com/danielperna84/hahomematic/assets/54898249/3da90199-e54e-4ff1-a20d-2bcb961f64f0)

**Reply by SukramJ** [reply]
Mir fehlt aktuell ein Testgerät das unterm Kanal noch mehrere Parameter hat. 
Ohne Unter-Parameter hab ich noch nichts gefunden.
Ich werde mir mal nächste Woche einen Test schreiben.

**Reply by Baxxy13** [reply]
Danke das du Dir das anguckst.
Es ging ja "früher" problemlos. Leider kann ich nicht sagen ab welcher Version (HA / Integration) das nun nicht mehr geht.

> Mir fehlt aktuell ein Testgerät das unterm Kanal noch mehrere Parameter hat

Das verstehe ich nicht. Mein Gerät hat 2 Kanäle. 
Vom Aufbau her ist es eigentlich Standard, wie z.B. ein STHO (ok, der hat mehr Kanäle aber :0 und :1) sind quasi identisch.
Nur das eben mein Gerät auf :1 mehr Datenpunkte hat, und speziell 2 "nonkonforme" deren Namen sich nicht an die Default-Nomenklatur" einer CCU halten.

Soll ich vielleicht mal nach eventuellen Fehlern im Log schauen wenn ich das Teil nochmal aus HA lösche und neu erkennen lasse?
Wenn ja welche Logger-Einstellungen wären richtig?

**Reply by SukramJ** [reply]
Ok, falsch ausgedrückt. Meine Parameter heißen `STATE` und die werden entfernt.

In den Logs wirst Du nichts finden.
Ggf. mal den Cache löschen und das Gerät aus HA entfernen. Danach neustarten.

**Reply by Baxxy13** [reply]
Mache ich immer so, hilft also leider nicht.

**Reply by SukramJ** [reply]
Ist doch eine Übersetzungsproblem.
Das Problem hat mit den Änderungen in HA für die übersetzbaren Entitätsnamen zu tun. Dadurch hat sich das Thema Namensgenerierung verkompliziert.
HAHomematic liefert die richtigen Namen, aber die werden z.Bsp. für `Taupunkt` nicht richtig umgesetzt.

Ich schau mir das nächste Woche mal an.

**Reply by SukramJ** [reply, marked as answer]
Die aktuelle Beta 1.51.0b0 zeigt den Parameter wieder im entitätsnamen an.

**Reply by Baxxy13** [reply]
Besten Dank. 
Es sieht auf HA-Seite nun alles wieder vollständig und korrekt aus.
Ich habe das gleich mal genutzt und die Namen auf der Zentrale angepasst.
Somit habe ich nun (nach der Re-Integration) korrekte und zusätzlich nicht mehr so ellenlange Entitäts-Namen.
